app/frontend/controllers/text_controller.js

```diff
@@ -713,16 +755,33 @@ class TextController {
     }
 
     if (isCache ||
-        listType != 'book' ||
-        listType == 'book' && append ||
-        !isInstantLoadingBook) {
+      listType != 'book' ||
+      listType == 'book' && append ||
+      !isInstantLoadingBook) {
 
       app_controller.optionsMenu.showOrHideSectionTitlesBasedOnOption(tabIndex);
-      await app_controller.initApplicationForVerseList(tabIndex);      
+      await app_controller.initApplicationForVerseList(tabIndex);
       uiHelper.hideTextLoadingIndicator();
     }
   }
 }
 
 module.exports = TextController;
 
+function filterNotesReferenceIds(verseNotes, startVerseNr, endVerseNr) {
+  const verseReferenceIds = Object.keys(verseNotes);
+  return verseReferenceIds.filter(verseReferenceId => {
+    const noteVerseNumber = parseInt(verseReferenceId.split('-')[2]);
+    return noteVerseNumber >= startVerseNr && (endVerseNr < startVerseNr || noteVerseNumber <= endVerseNr);
+  });
+}
+
+function filterNotesForVerses(verseNotes, startVerseNr, endVerseNr) {
```

Comment:
Wouldn't it make more sense to call it `getNotesForSection`. The function name does not indicate how we filter the notes.

Reply:
It doesn't "get notes" though. All notes come in `verseNotes` and returned only notes that are in range between `startVerseNr` and `endVerseNr`. I was thinking the name should be parallel to `Array.filter()`
Although, I guess you can call this "get".

Reply:
We can keep it this way, too!